clang/lib/CIR/Lowering/ThroughMLIR/LowerCIRToMLIR.cpp

```diff
@@ -488,12 +488,43 @@ class CIRBrOpLowering : public mlir::OpRewritePattern<mlir::cir::BrOp> {
   }
 };
 
+class CIRScopeOpLowering : public mlir::OpRewritePattern<mlir::cir::ScopeOp> {
+  using mlir::OpRewritePattern<mlir::cir::ScopeOp>::OpRewritePattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(mlir::cir::ScopeOp scopeOp,
+                  mlir::PatternRewriter &rewriter) const override {
+    // Empty scope: just remove it.
+    if (scopeOp.getRegion().empty()) {
+      rewriter.eraseOp(scopeOp);
+      return mlir::success();
+    }
+
+    for (auto &block : scopeOp.getRegion()) {
+      rewriter.setInsertionPointToEnd(&block);
+      auto yield = cast<mlir::cir::YieldOp>(block.getTerminator());
```

Comment:
Does every block in the scope have a yield? I recall that some blocks can have a return op. @bcardosolopes can you please confirm?

Reply:
Right, possible terminators are `YieldOp, ReturnOp, ThrowOp`

Reply:
Got it. The [doc](https://llvm.github.io/clangir/Dialect/ops.html#cirscope-cirscopeop) seems to be a little outdated.

Reply:
Created #265 to track this.